src/main/java/Tree/CartesianTree.java

```diff
@@ -0,0 +1,79 @@
+package Tree;
+
+import Node.BinaryNode;
+
+
+public class CartesianTree<T> extends BinaryTree<T> {
+    int[] sequence;
+    int length;
+
+    public CartesianTree() { current = null; }
+
+    public CartesianTree(int[] sequence, int length) {
+        current = null;
+        this.sequence = sequence;
+        this.length = length;
+    }
+
+    public void heapTree(Boolean maxHeap) { current = buildHeapTree(this.sequence, 0, length-1, maxHeap); }
+
+    protected BinaryNode<T> buildHeapTree(int[] seq, int begin, int end, Boolean isMaxHeap) {
+        if (begin > end) return null;
+        BinaryNode<T> node = null;
+        int index = 0;
+        Boolean isMax = false;
+
+        if (isMaxHeap) {
+            index = maxOrMin(seq, begin, end, true);
+            node = new BinaryNode<T>(seq[index], null);
+            isMax = true;
+        } else {
+            index = maxOrMin(seq, begin, end, false);
+            node = new BinaryNode<T>(seq[index], null);
+        }
+
+        if (begin == end) return node;
+
+        linkNodes(node, buildHeapTree(seq, begin, index-1, isMax), 'L');
+        if (node.getLeftNode() != null) { node.getLeftNode().setParentNode(node); }
```

Comment:
I thought `node.getLeftNode().setParentNode(node); ` is handled internally by linkNodes